ddtrace/internal/datadog/profiling/dd_wrapper/include/libdatadog_helpers.hpp

```diff
@@ -0,0 +1,131 @@
+    X(runtime_id, "runtime-id")                                                                                        \
+    X(profiler_version, "profiler_version")                                                                            \
+    X(profile_seq, "profile_seq")
+
+// Here there are two columns because the Datadog backend expects these labels
+// to have spaces in the names.
+#define EXPORTER_LABELS(X)                                                                                             \
+    X(exception_type, "exception type")                                                                                \
+    X(thread_id, "thread id")                                                                                          \
+    X(thread_native_id, "thread native id")                                                                            \
+    X(thread_name, "thread name")                                                                                      \
+    X(task_id, "task id")                                                                                              \
+    X(task_name, "task name")                                                                                          \
+    X(span_id, "span id")                                                                                              \
+    X(local_root_span_id, "local root span id")                                                                        \
+    X(trace_type, "trace type")                                                                                        \
+    X(trace_resource_container, "trace resource container")                                                            \
+    X(trace_endpoint, "trace endpoint")                                                                                \
+    X(class_name, "class name")                                                                                        \
+    X(lock_name, "lock name")
+
+#define X_ENUM(a, b) a,
+#define X_STR(a, b) b,
+
+enum class ExportTagKey
+{
+    EXPORTER_TAGS(X_ENUM) _Length
+};
+
+enum class ExportLabelKey
+{
+    EXPORTER_LABELS(X_ENUM) _Length
+};
+
+inline ddog_CharSlice
+to_slice(std::string_view str)
+{
+    return { .ptr = str.data(), .len = str.size() };
+}
+
+inline std::string
+err_to_msg(const ddog_Error* err, std::string_view msg)
+{
+    auto ddog_err = ddog_Error_message(err);
+    std::string err_msg;
+    return std::string{ msg } + " (" + err_msg.assign(ddog_err.ptr, ddog_err.ptr + ddog_err.len) + ")";
+}
+
+inline std::string_view
+to_string(ExportTagKey key)
+{
+    constexpr size_t num_keys = static_cast<size_t>(ExportTagKey::_Length);
+    constexpr std::array<std::string_view, num_keys> keys = { EXPORTER_TAGS(X_STR) };
+    constexpr std::string_view invalid = "";
+
+    if (static_cast<size_t>(key) >= num_keys)
+        return invalid;
+    return keys[static_cast<size_t>(key)];
```

Comment:
```suggestion
    size_t key_idx = static_cast<size_t>(key);

    if (key_idx) >= num_keys)
        return invalid;
    return keys[key_idx];
```

Reply:
Good call, factored it out though, sorry